packages/agents/src/agents/Chameleon/StandardRagBuilder.ts

```diff
@@ -0,0 +1,54 @@
+import { OpenAIEmbeddingAPI, LocalVectorDB } from "@evo-ninja/agent-utils";
+import { AgentContext } from "../../AgentContext";
+
+export class StandardRagBuilder<TItem> {
+  private _limit: number;
+  private _items: TItem[];
+  private _selector: (item: TItem) => string;
+
+  constructor(private readonly context: AgentContext) {
+  }
+
+  items(items: TItem[]): StandardRagBuilder<TItem> {
+    this._items = items;
+    return this;
+  }
+
+  limit(limit: number): StandardRagBuilder<TItem> {
+    this._limit = limit;
+    return this;
+  }
+
+  selector(selector: (item: any) => string): StandardRagBuilder<TItem> {
+    this._selector = selector;
+    return this;
+  }
+
+  async query(query: string): Promise<TItem[]> {
+    const embeddingApi = new OpenAIEmbeddingAPI(
+      this.context.env.OPENAI_API_KEY,
+      this.context.logger,
+      this.context.chat.tokenizer
+    );
+
+    const db = new LocalVectorDB(this.context.workspace, "testdb", embeddingApi);
```

Comment:
NIT: instead of constructing `StandardRagBuilder` with the `AgentContext`, it could be constructed with a `VectorDB` instance.

Reply:
Yea, I can see that, but don't want to spend time thinking about it :D